synnax/pkg/distribution/channel/lease_proxy.go

```diff
@@ -217,27 +213,24 @@ func (lp *leaseProxy) createGateway(
 		return err
 	}
 
-	numExternalToCreate := len(toCreate)
-	var internalCreatedKeys Keys
+	var externalCreatedKeys Keys
 	for _, ch := range toCreate {
-		if ch.Internal {
-			numExternalToCreate--
-			internalCreatedKeys = append(internalCreatedKeys, ch.Key())
+		if !ch.Internal {
+			externalCreatedKeys = append(externalCreatedKeys, ch.Key())
 		}
 	}
-	totalExternalChannels := LocalKey(numExternalToCreate) + lp.externalCounter.value()
-	if numExternalToCreate != 0 {
+	newExternalChannels := len(externalCreatedKeys)
+	totalExternalChannels := LocalKey(newExternalChannels) + lp.externalCounter.value()
+	if newExternalChannels != 0 {
 		if err = lp.IntOverflowCheck(ctx, types.Uint20(totalExternalChannels)); err != nil {
 			return err
 		}
 	}
-
-	_, err = lp.externalCounter.add(LocalKey(numExternalToCreate))
+	lp.external.Insert(externalCreatedKeys.Local()...)
```

Comment:
Careful here. If there are no external channels being created, `externalCreatedKeys` could be a nil reference to the slice, and the method could fail. You should pre-allocated `externalCreatedKeys`